### PR TITLE
upload e2e cloudcore and edgecore logs

### PR DESCRIPTION
**Pull request by gy95**
**What type of PR is this?**

**What this PR does / why we need it**:
upload e2e cloudcore and edgecore logs to help developers debug
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by wackxu** [comment]
lgtm

**Comment by wackxu** [comment]
lgtm

**Review by fisherxu — COMMENTED** [review]
Where the logs will be uploaded?

**Comment by gy95** [comment]
In the summary of the action, at the bottom of the web page.
For example: https://github.com/kubeedge/kubeedge/actions/runs/2418779085
This maybe benefit the debug when e2e tests fail and developers want to look up edgecore or cloudcore logs
Is it necessary to keep this? @fisherxu

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -170,6 +170,13 @@ jobs:
 
       - run: make e2e
 
+      - uses: actions/upload-artifact@v3
+        with:
+          name: my-artifact
+          path: |
+            /tmp/cloudcore.log
+            /tmp/edgecore.log
```
Comment (fisherxu) [inline .github/workflows/main.yaml:178]: > In the summary of the action, at the bottom of the web page.
For example: https://github.com/kubeedge/kubeedge/actions/runs/2418779085
This maybe benefit the debug when e2e tests fail and developers want to look up edgecore or cloudcore logs
Is it necessary to keep this? @fisherxu

Looks good, and seems the name can be `e2e-test-logs`? Other e2e tests like `keadm e2e`, we can also upload the logs?

**Comment by gy95** [comment]
```
Starting kubeedge... 1.11.0-alpha.0.2+aedcbf2d1ee549
Kubernetes version verification passed, KubeEdge installation will start...
Error: failed to exec 'bash -c cd /etc/kubeedge/crds/reliablesyncs && wget -k --no-check-certificate --progress=bar:force https://raw.githubusercontent.com/kubeedge/kubeedge/release-1.11/build/crds/reliablesyncs/cluster_objectsync_v1alpha1.yaml', err: --2022-06-02 07:17:41--  https://raw.githubusercontent.com/kubeedge/kubeedge/release-1.11/build/crds/reliablesyncs/cluster_objectsync_v1alpha1.yaml
execute keadm command failed:  failed to exec 'bash -c cd /etc/kubeedge/crds/reliablesyncs && wget -k --no-check-certificate --progress=bar:force https://raw.githubusercontent.com/kubeedge/kubeedge/release-1.11/build/crds/reliablesyncs/cluster_objectsync_v1alpha1.yaml', err: --2022-06-02 07:17:41--  https://raw.githubusercontent.com/kubeedge/kubeedge/release-1.11/build/crds/reliablesyncs/cluster_objectsync_v1alpha1.yaml
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 1[85](https://github.com/kubeedge/kubeedge/runs/6703808637?check_suite_focus=true#step:8:86).199.108.133, 185.199.111.133, 185.199.109.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2022-06-02 07:17:41 ERROR 404: Not Found.
```
error occurs in installing v1.11 crds, maybe related to new v1.11 alpha tag.

File: .github/workflows/main.yaml (hunk 1 above)
Comment (gy95) [reply .github/workflows/main.yaml:178]: keadm e2e logs will be logged by systemd, so we can only upload `e2e` logs

**Comment by gy95** [comment]
CC @fisherxu  only support upload e2e logs(due to both keadm_e2e and keadm_helm_e2e logs using systemd or helm, so not easy to upload those logs)
And other codes are to fix keadm e2e error(we have added v1.11-beta tag, so keadm will install v1.11 CRDs, but we haven't released v1.11, so I add a logic to keadm to compare kubeedge-version with the latest version, if the version is bigger than the latest version, we will choose the latest version to install CRDs) Is this OK? @fisherxu 
PTAL, the latter may block the new PRs.

**Comment by gy95** [comment]
CC @fisherxu  only support upload e2e logs(due to both keadm_e2e and keadm_helm_e2e logs using systemd or helm, so not easy to upload those logs)
And other codes are to fix keadm e2e error(we have added v1.11-beta tag, so keadm will install v1.11 CRDs, but we haven't released v1.11, so I add a logic to keadm to compare kubeedge-version with the latest version, if the version is bigger than the latest version, we will choose the latest version to install CRDs) Is this OK? @fisherxu 
PTAL, the latter may block the new PRs.

**Review by vincentgoat — APPROVED** [review]
/lgtm
We can't run e2e test successfully if this error hasn't been fixed.

File: keadm/cmd/keadm/app/cmd/util/common.go [hunk 2]
```diff
@@ -825,6 +825,18 @@ func downloadServiceFile(componentType types.ComponentType, version semver.Versi
...
+		// if the specified kubeedge version is greater than the latest version
+		// that means we haven't released the specified version, only occur in keadm e2e test
+		// we will use the latest version to download service file
+		latestVersion, _ := GetLatestVersion()
+		if v, err := semver.Parse(strings.TrimPrefix(latestVersion, "v")); err == nil {
+			if version.GT(v) {
+				strippedVersion = fmt.Sprintf("%d.%d", v.Major, v.Minor)
+			}
+		}
+		fmt.Printf("keadm will download version %s service file\n", strippedVersion)
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/util/common.go:838]: The `version` seems also got through `GetLatestVersion`, can we fix this issue in `Add2ToolsList` or in `GetLatestVersion`?

Comment (gy95) [reply keadm/cmd/keadm/app/cmd/util/common.go]: seems not proper, we only use the latest version to download service file and CRDs, but we also need the version to process release package, this value should use the exact version but not the latest version.

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**